packages/workbox-routing/lib/Route.mjs

```diff
@@ -0,0 +1,170 @@
+/*
+ Copyright 2017 Google Inc. All Rights Reserved.
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+import core from 'workbox-core';
+
+import {defaultMethod, validMethods} from './constants.mjs';
+
+/**
+ * This is the definition of the `match` callback passed into the
+ * `Route` constructor.
+ *
+ * This callback is used to determine if a new `fetch` event can be served
+ * by this `Route`. Returning a truthy value indicates that this `Route` can
+ * handle this `fetch` event. Return `null` if this shouldn't match against
+ * the `fetch` event.
+ *
+ * If you do return a truthy value, the object will be passed to the
+ * Route's `handler` (see the
+ * [Route Constructor]{@link module:workbox-routing.Route}).
+ *
+ * @callback Route~matchCallback
+ * @param {Object} input
+ * @param {URL} input.url The request's URL.
+ * @param {FetchEvent} input.event The event that triggered the `fetch` handler.
+ * @return {Object|null} To signify a match, return a truthy value, otherwise
+ * return null if the route shouldn't match. If you return an Object with
```

Comment:
Nit: Could you change "If you return an Object with contents it will be passed to the `handler` in the `Route` constructor." to "If you return an Object with contents, it will be passed to the Route's `handler` (see the [Route Constructor]{@link module:workbox-routing.Route}).

Reply:
I'm going to go with `{@link module:workbox-routing.Route~handlerCallback}` instead, since that's what gets passed the parameter.